### PR TITLE
Document the deprecation policy of HTML tags.

### LINKED ISSUE #1492: Deprecated strike tag in custom html
**Link to problem area**:
https://spec.matrix.org/v1.6/client-server-api/#mroommessage-msgtypes

**Issue**
The strongly suggested set of "HTML tags to permit" mention deprecated [strike](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/strike) tag. It should be replaced with [s](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/s) tag.

### PR DESCRIPTION
**Pull request by zecakeh**
~~Replace it with `s` or `del`, as per [MSC4077](https://github.com/matrix-org/matrix-spec-proposals/pull/4077).~~

It also adds an info box as suggested in https://github.com/matrix-org/matrix-spec-proposals/pull/4077#discussion_r1393595760.

~~Closes #1492.~~

Preview: https://pr1732--matrix-spec-previews.netlify.app

### CODE REVIEW

**Comment by zecakeh** [comment]
Woops, I just saw that #1629 was reopened for this. Should I keep this open only for the deprecation info box?

File: content/client-server-api/modules/instant_messaging.md [hunk 1]
```diff
@@ -37,9 +37,24 @@ HTML injection, and similar attacks. The strongly suggested set of HTML
 tags to permit, denying the use and rendering of anything else, is:
 `font`, `del`, `h1`, `h2`, `h3`, `h4`, `h5`, `h6`, `blockquote`, `p`,
 `a`, `ul`, `ol`, `sup`, `sub`, `li`, `b`, `i`, `u`, `strong`, `em`,
-`strike`, `code`, `hr`, `br`, `div`, `table`, `thead`, `tbody`, `tr`,
+`strike`, `s`, `code`, `hr`, `br`, `div`, `table`, `thead`, `tbody`, `tr`,
 `th`, `td`, `caption`, `pre`, `span`, `img`, `details`, `summary`.
 
+
+{{% boxes/note %}}
+HTML features MAY be deprecated and replaced by their modern equivalent without
```
Comment (turt2live) [inline content/client-server-api/modules/instant_messaging.md:45]:
```suggestion
{{% added-in v="1.10" %}}
HTML features MAY be deprecated and replaced by their modern equivalent without
```

or whatever the syntax is

File: content/client-server-api/modules/instant_messaging.md [hunk 2]
```diff
@@ -37,9 +37,24 @@ HTML injection, and similar attacks. The strongly suggested set of HTML
...
+requiring a [Spec Change Proposal](/proposals) when they are deprecated in the
+WHATWG HTML Living Standard.
+{{% /boxes/note %}}
+
+{{% boxes/note %}}
+{{% changed-in v="1.10" %}}
+
+The `strike` tag is deprecated. Clients MUST stop sending new messages using
+this tag and replace it with `s` or `del`.
+{{% /boxes/note %}}
```
Comment (turt2live) [inline content/client-server-api/modules/instant_messaging.md:55]:
```suggestion
```

The MSC does not permit this strong of language. What we can do is simply remove `strike`, like in https://github.com/matrix-org/matrix-spec/pull/1629

File: content/client-server-api/modules/instant_messaging.md [hunk 3]
```diff
@@ -37,9 +37,24 @@ HTML injection, and similar attacks. The strongly suggested set of HTML
 tags to permit, denying the use and rendering of anything else, is:
 `font`, `del`, `h1`, `h2`, `h3`, `h4`, `h5`, `h6`, `blockquote`, `p`,
```
[the remark this answers is not in the document]
Comment (turt2live) [reply content/client-server-api/modules/instant_messaging.md:38]: as long as we can already represent it under the existing specification. If we can't, it's not deprecated on our side without an MSC.

**Review by turt2live — CHANGES_REQUESTED** [review]
I've accepted https://github.com/matrix-org/matrix-spec/pull/1629 to cover the `strike`->`s` change, which I believe should allow this PR to handle the addition of a note about HTML specification changes (which the changelog will need to mention).

File: content/client-server-api/modules/instant_messaging.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (zecakeh) [reply content/client-server-api/modules/instant_messaging.md:38]: To clarify why the deprecation is possible:

The `font` tag doesn't change the representation of the content without attributes, like the `span` tag. It allows the attributes `data-mx-bg-color`, `data-mx-color` and `color`. The two first attributes are available on `span`, and `color` and `data-mx-color` have the same effect.

So, HTML like that:

```html
<font data-mx-bg-color="#FF0000" color="#00FF00">Some green text on a red background</font>
```

Can also be written like that:

```html
<span data-mx-bg-color="#FF0000" data-mx-color="#00FF00">Some green text on a red background</span>
```

File: content/client-server-api/modules/instant_messaging.md (hunk 1 above)
Comment (zecakeh) [reply content/client-server-api/modules/instant_messaging.md:53]: Applied

File: content/client-server-api/modules/instant_messaging.md (hunk 2 above)
Comment (zecakeh) [reply content/client-server-api/modules/instant_messaging.md]: Removed

File: content/client-server-api/modules/instant_messaging.md [hunk 4]
```diff
@@ -48,6 +48,13 @@ tags to permit, denying the use and rendering of anything else, is:
 `s`, `code`, `hr`, `br`, `div`, `table`, `thead`, `tbody`, `tr`,
 `th`, `td`, `caption`, `pre`, `span`, `img`, `details`, `summary`.
 
+{{% boxes/note %}}
+{{% added-in v="1.10" %}}
+HTML features MAY be deprecated and replaced by their modern equivalent without
+requiring a [Spec Change Proposal](/proposals) when they are deprecated in the
+WHATWG HTML Living Standard.
```
Comment (uhoreg) [inline content/client-server-api/modules/instant_messaging.md:55]: link to the standard

```suggestion
[WHATWG HTML Living Standard](https://html.spec.whatwg.org/multipage/).
```

Comment (zecakeh) [reply content/client-server-api/modules/instant_messaging.md]: Applied

File: changelogs/client_server/newsfragments/1732.clarification [hunk 5]
```diff
@@ -0,0 +1 @@
+Document the deprecation policy of HTML tags.
```
Comment (turt2live) [inline changelogs/client_server/newsfragments/1732.clarification:1]:
```suggestion
Document the deprecation policy of HTML tags, as per [MSC4077](https://github.com/matrix-org/matrix-spec-proposals/pull/4077).
```

**Review by turt2live — APPROVED** [review]
Thanks!

**Pull request merged**